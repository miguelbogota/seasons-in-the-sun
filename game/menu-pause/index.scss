@use './menu-pause' as menu-pause;

@mixin apply {
  @include menu-pause.apply;
}
