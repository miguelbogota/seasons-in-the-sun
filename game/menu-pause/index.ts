export { MenuPause } from './menu-pause.component';
