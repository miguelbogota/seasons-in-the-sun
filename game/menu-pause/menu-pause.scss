@mixin apply {
  .menu-pause {
    position: fixed;
    inset: 0;
    z-index: var(--navigation);
    background-color: #7a7a7a7d;

    & > .menu {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      width: 100%;
      max-width: 400px;
      padding: var(--space-4);
      background-color: var(--background-sheet);
      border-radius: var(--radius-normal);
      box-shadow: 0 0 10px 0 #00000080;

      button {
        width: 100%;
        padding: var(--space-2);
        background-color: var(--background-main);
        text-align: center;
        margin-bottom: 1rem;
        border-radius: var(--radius-normal);
        font-size: var(--size-5);
        font-weight: var(--weight-bold);
        color: var(--primary);
        cursor: pointer;
        transition:
          background-color 0.2s,
          border 0.2s,
          color 0.2s;

        &:last-child {
          margin-bottom: 0;
        }
      }
    }
  }
}
