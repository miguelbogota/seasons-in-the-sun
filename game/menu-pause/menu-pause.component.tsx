'use client';

import { emitCameraControlEvent } from '@app/game/player/camera-control';
import { useGameState } from '@app/game/state';

/**
 * Component with the pause menu of the game.
 */
export function MenuPause() {
  const state = useGameState();

  return (
    state.isPlaying &&
    state.isPaused && (
      <div className="menu-pause">
        <div className="menu">
          <button
            disabled={!state.canUnpause}
            onClick={() => {
              state.pauseGame();
              emitCameraControlEvent('lock-controls', null);
            }}
          >
            Continue
          </button>
          <button>Options</button>
          <button onClick={() => state.exitGame()}>Quit</button>
        </div>
      </div>
    )
  );
}
