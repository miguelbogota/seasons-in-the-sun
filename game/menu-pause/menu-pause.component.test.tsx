import { render } from '@testing-library/react';

import { MenuPause } from './menu-pause.component';

describe('Game / MenuPause / Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<MenuPause />);
    expect(container).toMatchSnapshot();
  });
});
