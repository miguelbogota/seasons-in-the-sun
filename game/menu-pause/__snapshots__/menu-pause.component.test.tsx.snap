// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game / MenuPause / Component > should match snapshot 1`] = `<div />`;
